src/FrequencyGovernor.hpp

```diff
@@ -28,6 +28,14 @@ namespace geopm
             ///        the vector to pass to adjust_platform().
             /// @return The domain with which frequency will be governed.
             virtual int frequency_domain_type(void) const = 0;
+            /// @brief Set the domain type of frequency control that will
+            ///        be used in adjust_platform(). Must be called before
+            ///        init_platform_io().
```

Comment:
There doesn't seem to be any issue with a user calling init_platform_io(), checking the domain, calling set_domain_type, and then re-calling init_platform_io().  As written this implies you have to call set_domain_type() before any call to init_platform_io().

Rewording for clarity might help.